plugins/modules/source_control/gitlab/gitlab_group_members.py

```diff
@@ -0,0 +1,235 @@
+            group.members.delete(gitlab_user_id)
+
+        except (gitlab.exceptions.GitlabDeleteError) as e:
+            self._module.fail_json(
+                msg="Failed to remove member from GitLab group, ID %s: %s" % (gitlab_group_id, e))
+
+
+def main():
+    argument_spec = basic_auth_argument_spec()
+    argument_spec.update(dict(
+        api_token=dict(type='str', required=True, no_log=True),
+        gitlab_group=dict(type='str', required=True),
+        gitlab_user=dict(type='str', required=True),
+        state=dict(type='str', default='present', choices=['present', 'absent']),
+        access_level=dict(type='str', required=False, choices=['guest', 'reporter', 'developer', 'maintainer', 'owner'])
+    ))
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        mutually_exclusive=[
+            ['api_username', 'api_token'],
+            ['api_password', 'api_token'],
+        ],
+        required_together=[
+            ['api_username', 'api_password'],
+        ],
+        required_one_of=[
+            ['api_username', 'api_token']
+        ],
+        supports_check_mode=True,
+    )
+
+    if not HAS_PY_GITLAB:
+        module.fail_json(msg=missing_required_lib('python-gitlab', url='https://python-gitlab.readthedocs.io/en/stable/'), exception=GITLAB_IMP_ERR)
+
+    gitlab_group = module.params['gitlab_group']
+    gitlab_user = module.params['gitlab_user']
+    state = module.params['state']
+    access_level = module.params['access_level']
+
+    # convert access level string input to int
+    if access_level:
+        access_level_int = {
+            'guest': gitlab.GUEST_ACCESS,
+            'reporter': gitlab.REPORTER_ACCESS,
+            'developer': gitlab.DEVELOPER_ACCESS,
+            'maintainer': gitlab.MAINTAINER_ACCESS,
+            'owner': gitlab.OWNER_ACCESS
+        }
+
+        access_level = access_level_int[access_level]
+
+    # connect to gitlab server
+    gl = gitlabAuthentication(module)
+
+    group = GitLabGroup(module, gl)
+
+    gitlab_user_id = group.get_user_id(gitlab_user)
+    gitlab_group_id = group.get_group_id(gitlab_group)
+
+    # group doesn't exist
+    if not gitlab_group_id:
+        module.fail_json(msg="group '%s' not found." % gitlab_group)
+
+    # user doesn't exist
+    if not gitlab_user_id:
+        if state == 'absent':
+            module.exit_json(changed=False, result="user '%s' not found, and thus also not part of the group" % gitlab_user)
+        else:
+            module.fail_json(msg="user '%s' not found." % gitlab_user)
+
+    is_user_a_member = group.is_user_a_member(gitlab_group_id, gitlab_user_id)
+
+    # check if the user is a member in the group
+    if not is_user_a_member:
+        if state == 'present':
+
+            if not access_level:
+                module.fail_json(msg="Access level must be entered for adding the member, '%s', to the GitLab group, '%s'." % (gitlab_user, gitlab_group))
```

Comment:
```suggestion
```
The `required_if` will take care of this. (It also prevents the module from running with `state=present` if the user is already a member of the group, but that's better in general.)